### PR TITLE
1st draft of new article

### PR DESCRIPTION
**Pull request by doctordns**
# PR Summary
A new article on ways to get PowerShell output to a file.

This PR is in response to #13.

## PR Checklist

- Do you want this post to be published on a specific date?: Y/N [Specify date]:
- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] Includes content related an open issue
  - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: Posts/2021/05/tfl-output_as_string.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+## The alternative methods
+
+There are (at least) four ways to get output to a file.
+You can use any or all of:
+
+* `*-Content` cmdlets
+* `Out-File` cmdlet
+* Redirection operators
+* .NET classes
```
Comment (sdwheeler) [inline Posts/2021/05/tfl-output_as_string.md:36]: This section is repetitive. I suggest moving line 23-25 to the list below.

File: Posts/2021/05/tfl-output_as_string.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+One of PowerShell's great features is the way it automatically formats output.
+You type a command - PowerShell gives you the output it thinks you want.
+If the default output is not what you need, use the formatting cmdlets like `Format-Table` and `Format-List` to get what you want.
+But sometimes, what you want is getting output to a file, not to the console.
+You might want to run a command or script that outputs information to a file and sends this file via email or possibly FTP.
+Or, you might want to view it in a text editor or print it out later.
+
+Once you have created the code (script, fragment, or a single command) that creates the output you need, you can use several techniques to send that output to a file.
+The easiest way to do this is to pipe the output to `Add-Content`, `Set-Content`, or `Out-File`.
+Then, there is the legacy approach, using the **>** or **>>** characters.
```
Comment (sdwheeler) [inline Posts/2021/05/tfl-output_as_string.md:24]:
```suggestion
Then, there is the legacy approach, using the redirection operators, `>` or `>>` .
```

**Review by sdwheeler — CHANGES_REQUESTED** [review]

File: Posts/2021/05/tfl-output_as_string.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+* `Set-Content` - writes new content which replaces the content in a file.
+
+The two cmdlets you use to send command or script output to a file are `Set-Content` and `Add-Content`.
+Both cmdlets convert the objects you pass in the pipeline to strings, and then output these strings to the specified file.
+A very important point here - if you pass either cmdlet a non-string object, these cmdlets use each object's **ToString()** method to convert the object to a string before outputting it to the file.
+For example:
+
+```powershell-console
+PS> Get-Process -Name pwsh | Set-Content -Path C:\Foo\AAA.txt
+PS> Get-Content -Path C:\Foo\AAA.txt
```
[the remark this answers is not in the document]
Comment (doctordns) [reply Posts/2021/05/tfl-output_as_string.md:60]: fixed.

File: Posts/2021/05/tfl-output_as_string.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+For example, you could create a string, then add to it repeatedly in a script, finally outputting the report string to a file.
+You can see the basic approach to building up a report in [this script that creates a Hyper-V VM summary report](https://github.com/doctordns/Wiley20/blob/master/10%20-%20Reporting/10.8%20-%20Creating%20a%20Hyper-V%20Status%20Report.ps1).
+
+You can improve the output from `Set-Content` by using `Out-String`, like this:
+
+```powershell-console
+PS> # Get Powershell processes, convert to string, then output to a file
+PS> Get-Process -Name pwsh |
+      Out-String |
+        Set-Content .\Process.txt
```
[the remark this answers is not in the document]
Comment (doctordns) [reply Posts/2021/05/tfl-output_as_string.md:92]: fixedf

**Review by sdwheeler — APPROVED** [review]

**Pull request merged**